src/sardana/macroserver/msmacromanager.py

```diff
@@ -1290,9 +1294,31 @@ def __runXMLMacro(self, xml):
     _runXMLMacro = __runXMLMacro
 
     def runMacro(self, macro_obj):
+
         name = macro_obj._getName()
         desc = macro_obj._getDescription()
         door = self.door
+        self.macro_obj = macro_obj
```

Comment:
I think this is dangerous. We can not maintain a reference to the macro_obj, otherwise the memory could not be freed after the macro execution. The macro_pointer reference and the environment variable `PreserveMacroObj` is related to that.

Reply:
Not needed any more (it was only used in the call done in jobEnded, now moved to runMacro). I remove it.